addons/stock/models/stock_move_line.py

```diff
@@ -860,3 +860,47 @@ def action_open_reference(self):
             'views': [[False, "form"]],
             'res_id': self.id,
         }
+
+    def action_revert_inventory(self):
+        move_vals = []
+        processed_move_line = self.env['stock.move.line']
+        for move_line in self:
+            from_inventory_location = move_line.location_id.usage == 'inventory' and move_line.location_id or False
```

Comment:
- You can check if the `move_line` `is_inventory`, so it's easier to get the correct location:
```python
if not move_line.is_inventory:
    continue
```
- By search a quant, your method doesn't work if the inventory you try to reverse set the quantity to 0 and the quant was deleted in the meantime.
Can you create a new move with the right values (copied from the existing move but with inversed location and with the right name) without the need of the quant ?

Reply:
okay let me try